File: flutter_baidu_yingyan_trace-1.0.0/lib/src/model/fence/fence_alarm_push_info.dart
import 'package:flutter_baidu_yingyan_trace/src/model/coord_type.dart';
import 'package:flutter_baidu_yingyan_trace/src/model/fence/fence_types.dart';
import 'package:flutter_baidu_yingyan_trace/src/model/lat_lng.dart';

/// 本地、服务端围栏报警推送消息
class FenceAlarmPushInfo {
  /// 围栏编号
  int ?fenceId;

  /// 围栏名称
  String ?fenceName;

  /// 监控对象
  String ?monitoredPerson;

  /// 监控动作
  MonitoredAction ?monitoredAction;

  /// 当前报警点轨迹信息
  AlarmPoint ?currentPoint;

  /// 上一个轨迹点信息
  AlarmPoint ?prePoint;

  FenceAlarmPushInfo.fromMap(Map map) : assert(map != null) {
    fenceId = map['fenceId'];
    fenceName = map['fenceName'];
    monitoredPerson = map['monitoredPerson'];
    if (map['monitoredAction'] != null) {
      int value = map['monitoredAction'] as int;
      if (value >= 0) {
        monitoredAction = MonitoredAction.values[value];
      }
    }

    currentPoint = map['currentPoint'] == null
        ? null
        : AlarmPoint.fromMap(map['currentPoint'] as Map);
    prePoint = map['prePoint'] == null
        ? null
        : AlarmPoint.fromMap(map['prePoint'] as Map);
  }

  Map toMap() {
    return {
      'fenceId': this.fenceId,
      'fenceName': this.fenceName,
      'monitoredPerson': this.monitoredPerson,
      'monitoredAction': this.monitoredAction?.index,
      'currentPoint': this.currentPoint?.toMap(),
      'prePoint': this.prePoint?.toMap(),
    };
  }
}

/// 围栏报警时轨迹点信息
class AlarmPoint {
  /// 坐标经纬度信息
  LatLng ?location;

  /// 坐标类型
  CoordType ?coordType;

  /// 定位时间（单位：秒）
  int ?locTime;

  /// 上传到服务端时间（只在查询服务端围栏历史报警信息时，该字段有值）
  int ?createTime;

  /// 定位精度（单位：米）
  double ?radius;

  AlarmPoint.fromMap(Map map) : assert(map != null) {
    location =
        map['location'] == null ? null : LatLng.fromMap(map['location'] as Map);
    if (map['coordType'] != null) {
      int value = map['coordType'] as int;
      if (value >= 0 && value <= 2) {
        coordType = CoordType.values[value];
      }
    }
    locTime = map['locTime'];
    createTime = map['createTime'];
    radius = map['radius'];
  }

  Map toMap() {
    return {
      'location': this.location?.toMap(),
      'coordType': this.coordType?.index,
      'locTime': this.locTime,
      'createTime': this.createTime,
      'radius': this.radius
    };
  }
}

File: flutter_baidu_yingyan_trace-1.0.0/android/src/main/java/com/baidu/flutter/trace/model/fence/FenceAlarmPushInfo.java
package com.baidu.flutter.trace.model.fence;

/**
 * 本地、服务端围栏报警推送消息
 *
 * @author baidu
 */
public class FenceAlarmPushInfo {

    /**
     * 围栏编号
     */
    public long fenceId;

    /**
     * 围栏名称
     */
    public String fenceName;

    /**
     * 监控对象
     */
    public String monitoredPerson;

    /**
     * 监控动作
     */
    public int monitoredAction;

    /**
     * 当前报警点轨迹信息
     */
    public AlarmPoint currentPoint;

    /**
     * 上一个轨迹点信息
     */
    public AlarmPoint prePoint;

    /**
     * 获取围栏编号
     *
     * @return
     */
    public long getFenceId() {
        return fenceId;
    }

    public void setFenceId(long fenceId) {
        this.fenceId = fenceId;
    }

    /**
     * 获取围栏名称
     *
     * @return
     */
    public String getFenceName() {
        return fenceName;
    }

    public void setFenceName(String fenceName) {
        this.fenceName = fenceName;
    }

    /**
     * 获取监控对象
     *
     * @return
     */
    public String getMonitoredPerson() {
        return monitoredPerson;
    }

    public void setMonitoredPerson(String monitoredPerson) {
        this.monitoredPerson = monitoredPerson;
    }

    /**
     * 获取监控动作
     *
     * @return
     */
    public int getMonitoredAction() {
        return monitoredAction;
    }

    public void setMonitoredAction(int monitoredAction) {
        this.monitoredAction = monitoredAction;
    }

    /**
     * 获取当前报警点轨迹信息
     *
     * @return
     */
    public AlarmPoint getCurrentPoint() {
        return currentPoint;
    }

    public void setCurrentPoint(AlarmPoint currentPoint) {
        this.currentPoint = currentPoint;
    }

    /**
     * 获取上一个轨迹点信息
     *
     * @return
     */
    public AlarmPoint getPrePoint() {
        return prePoint;
    }

    public void setPrePoint(AlarmPoint prePoint) {
        this.prePoint = prePoint;
    }

    public FenceAlarmPushInfo() {
        super();
    }

    /**
     * @param fenceId         围栏编号
     * @param fenceName       围栏名称
     * @param monitoredPerson 监控对象
     * @param monitoredAction 监控动作
     * @param currentPoint    当前报警点轨迹信息
     * @param prePoint        上一个轨迹点信息
     */
    public FenceAlarmPushInfo(long fenceId, String fenceName, String monitoredPerson,
                              int monitoredAction,
                              com.baidu.trace.api.fence.AlarmPoint currentPoint,
                              com.baidu.trace.api.fence.AlarmPoint prePoint) {
        super();
        this.fenceId = fenceId;
        this.fenceName = fenceName;
        this.monitoredPerson = monitoredPerson;
        this.monitoredAction = monitoredAction;
        if (currentPoint != null) {
            this.currentPoint = AlarmPoint.fromSDKObject(currentPoint);
        }

        if (prePoint != null) {
            this.prePoint = AlarmPoint.fromSDKObject(currentPoint);
        }
    }
    
    public static FenceAlarmPushInfo fromSDKObject(com.baidu.trace.api.fence.FenceAlarmPushInfo fenceAlarmPushInfo) {
        if (fenceAlarmPushInfo == null) {
            return null;
        }
        FenceAlarmPushInfo fenceAlarmPushInfoTmp = new FenceAlarmPushInfo();
        fenceAlarmPushInfoTmp.setFenceId(fenceAlarmPushInfo.getFenceId());
        fenceAlarmPushInfoTmp.setFenceName(fenceAlarmPushInfo.getFenceName());
        fenceAlarmPushInfoTmp.setMonitoredAction(fenceAlarmPushInfo.getMonitoredAction().ordinal());
        fenceAlarmPushInfoTmp.setMonitoredPerson(fenceAlarmPushInfo.getMonitoredPerson());
        fenceAlarmPushInfoTmp.setCurrentPoint(AlarmPoint.fromSDKObject(fenceAlarmPushInfo.getCurrentPoint()));
        fenceAlarmPushInfoTmp.setPrePoint(AlarmPoint.fromSDKObject(fenceAlarmPushInfo.getPrePoint()));
        return fenceAlarmPushInfoTmp;
    }

    @Override
    public String toString() {
        return "FenceAlarmPushInfo [fenceId=" + fenceId + ", fenceName=" + fenceName
                + ", monitoredPerson="
                + monitoredPerson + ", monitoredAction=" + monitoredAction + ", currentPoint="
                + currentPoint
                + ", prePoint=" + prePoint + "]";
    }

}
